editor/editor_node.cpp

```diff
@@ -2657,22 +2657,36 @@ void EditorNode::_menu_option_confirm(int p_option, bool p_confirmed) {
 		case FILE_CLOSE_ALL_AND_RELOAD_CURRENT_PROJECT: {
 			if (!p_confirmed) {
 				tab_closing_idx = _next_unsaved_scene(false);
-				_scene_tab_changed(tab_closing_idx);
+				if (tab_closing_idx == -1) {
+					tab_closing_idx = -2; // Only external resources are unsaved.
```

Comment:
Could maybe use a local bool instead of using a new magic number in an unrelated index?

Reply:
It can't be a local bool, because the value is also used elsewhere. Re-using the variable was simpler to do and is less error-prone than adding a new variable and making sure it's used where necessary.

Reply:
But outside this scope, does it serve a purpose and is it properly handled that `tab_closing_idx` stays `-2`?

What I meant with a local bool would be this:
```
tab_closing_idx = _next_unsaved_scene(false);
bool ext_res_only = false;
if (tab_closing_idx == -1) {
	ext_res_only = true;
} else {
	_scene_tab_changed(tab_closing_idx);
}
```

Reply:
Yes, here:
https://github.com/godotengine/godot/blob/8b79a19426c7f5eab04caa573866276c76c70871/editor/editor_node.cpp#L1958-L1960

Reply:
That's pretty smelly :(

But well, I guess this can be revisited in the future.